presto-docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -0,0 +1,74 @@
+=============================
+Secure Internal Communication
+=============================
+
+The Presto cluster can be configured to use secured communication. Communication
+between Presto nodes can be secured with encryption, authentication and authorization.
+
+.. Note: At this step only encryption is supported
+
+Encrypted Internal Communication Configuration
+----------------------------------------------
+
+Encryption is configured in `config.properties` file.
+Encryption properties are identical for all of the presto node roles (coordinator, worker)
+and must be set on every node in the cluster.
+
+Disable HTTP endpoint. Leaving HTTP endpoint enabled leads to a security hole.
+
+.. code-block:: none
+
+    http-server.http.enabled=false
+
+Specify node's domains. Every node must have it's own domain configured.
+Domains must be in the same parent domain to be able to easy generate custom SSL \
```

Comment:
For the simplicity i suggest above to modify only `config.properties`. We can implement auto-domain assignment in `presto-admin`, and modify `node.properties` there.